internal/authz/authz.go

```diff
@@ -0,0 +1,205 @@
+package authz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/authcontext"
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+const (
+	// API methods.
+	ReadAuthorizationModel  = "ReadAuthorizationModel"
```

Comment:
Maybe we should prefix these and use them as enums?

e.g.
```go
type APIMethod string

const (
	APIMethodReadAuthorizationModel APIMethod = "ReadAuthorizationModels"
	APIMethodRead    APIMethod = "Read"
  ...
)
```

Same for the relations below
```go
type FgaPermission string

const (
	FgaPermissionCanCallReadAuthorizationModels FgaPermission = "can_call_read_authorization_models"
  ...
)
```

And the types
```go
type FgaTypeName string

const (
	FgaTypeNameStore FgaTypeName = "store"
	FgaTypeNameApplication FgaTypeName = "application"
  ...
)

```

Reply:
Using the type name in the variable name is redundant. The Go standard library, and myself, prefer to just prefix constant names.
```go
// example: net/http/method.go
const (
	MethodGet     = "GET"
	MethodHead    = "HEAD"
	MethodPost    = "POST"
	MethodPut     = "PUT"
	MethodPatch   = "PATCH" // [RFC 5789](https://rfc-editor.org/rfc/rfc5789.html)
	MethodDelete  = "DELETE"
	MethodConnect = "CONNECT"
	MethodOptions = "OPTIONS"
	MethodTrace   = "TRACE"
)
```